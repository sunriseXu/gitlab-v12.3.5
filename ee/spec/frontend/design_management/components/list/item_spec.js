import { createLocalVue, shallowMount } from '@vue/test-utils';
import VueRouter from 'vue-router';
import Item from 'ee/design_management/components/list/item.vue';

const localVue = createLocalVue();
localVue.use(VueRouter);
const router = new VueRouter();

describe('Design management list item component', () => {
  let wrapper;

  function createComponent(notesCount = 1, event = 'NONE') {
    wrapper = shallowMount(Item, {
      sync: false,
      localVue,
      router,
      propsData: {
        id: 1,
        name: 'test',
        image: 'http://via.placeholder.com/300',
        event,
        notesCount,
        updatedAt: '01-01-2019',
      },
      stubs: ['router-link'],
    });
  }

  afterEach(() => {
    wrapper.destroy();
  });

  it('renders item with single comment', () => {
    createComponent();

    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders item with multiple comments', () => {
    createComponent(2);

    expect(wrapper.element).toMatchSnapshot();
  });

  it('hides comment count', () => {
    createComponent(0);

    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders item with correct status icon for modification event', () => {
    createComponent(0, 'MODIFICATION');

    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders item with correct status icon for deletion event', () => {
    createComponent(0, 'DELETION');

    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders item with correct status icon for creation event', () => {
    createComponent(0, 'CREATION');

    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders item with no status icon for none event', () => {
    createComponent(0, 'NONE');

    expect(wrapper.element).toMatchSnapshot();
  });
});
