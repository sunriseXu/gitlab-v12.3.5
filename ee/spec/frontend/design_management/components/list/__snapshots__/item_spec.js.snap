// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Design management list item component hides comment count 1`] = `
<router-link-stub
  class="card cursor-pointer text-plain js-design-list-item design-list-item"
  to="[object Object]"
>
  <div
    class="card-body p-0 d-flex align-items-center overflow-hidden position-relative"
  >
    <!---->
     
    <img
      alt="test"
      class="block ml-auto mr-auto mw-100 mh-100 design-img"
      src="http://via.placeholder.com/300"
    />
  </div>
   
  <div
    class="card-footer d-flex w-100"
  >
    <div
      class="d-flex flex-column str-truncated-100"
    >
      <span
        class="bold str-truncated-100"
      >
        test
      </span>
       
      <span
        class="str-truncated-100"
      >
        
        Updated 
        <timeago-stub
          cssclass=""
          time="01-01-2019"
          tooltipplacement="bottom"
        />
      </span>
    </div>
     
    <!---->
  </div>
</router-link-stub>
`;

exports[`Design management list item component renders item with correct status icon for creation event 1`] = `
<router-link-stub
  class="card cursor-pointer text-plain js-design-list-item design-list-item"
  to="[object Object]"
>
  <div
    class="card-body p-0 d-flex align-items-center overflow-hidden position-relative"
  >
    <div
      class="design-event position-absolute"
    >
      <span
        aria-label="Added in this version"
        title="Added in this version"
      >
        <icon-stub
          cssclasses="text-success-500"
          name="file-addition-solid"
          size="18"
        />
      </span>
    </div>
     
    <img
      alt="test"
      class="block ml-auto mr-auto mw-100 mh-100 design-img"
      src="http://via.placeholder.com/300"
    />
  </div>
   
  <div
    class="card-footer d-flex w-100"
  >
    <div
      class="d-flex flex-column str-truncated-100"
    >
      <span
        class="bold str-truncated-100"
      >
        test
      </span>
       
      <span
        class="str-truncated-100"
      >
        
        Updated 
        <timeago-stub
          cssclass=""
          time="01-01-2019"
          tooltipplacement="bottom"
        />
      </span>
    </div>
     
    <!---->
  </div>
</router-link-stub>
`;

exports[`Design management list item component renders item with correct status icon for deletion event 1`] = `
<router-link-stub
  class="card cursor-pointer text-plain js-design-list-item design-list-item"
  to="[object Object]"
>
  <div
    class="card-body p-0 d-flex align-items-center overflow-hidden position-relative"
  >
    <div
      class="design-event position-absolute"
    >
      <span
        aria-label="Deleted in this version"
        title="Deleted in this version"
      >
        <icon-stub
          cssclasses="text-danger-500"
          name="file-deletion-solid"
          size="18"
        />
      </span>
    </div>
     
    <img
      alt="test"
      class="block ml-auto mr-auto mw-100 mh-100 design-img"
      src="http://via.placeholder.com/300"
    />
  </div>
   
  <div
    class="card-footer d-flex w-100"
  >
    <div
      class="d-flex flex-column str-truncated-100"
    >
      <span
        class="bold str-truncated-100"
      >
        test
      </span>
       
      <span
        class="str-truncated-100"
      >
        
        Updated 
        <timeago-stub
          cssclass=""
          time="01-01-2019"
          tooltipplacement="bottom"
        />
      </span>
    </div>
     
    <!---->
  </div>
</router-link-stub>
`;

exports[`Design management list item component renders item with correct status icon for modification event 1`] = `
<router-link-stub
  class="card cursor-pointer text-plain js-design-list-item design-list-item"
  to="[object Object]"
>
  <div
    class="card-body p-0 d-flex align-items-center overflow-hidden position-relative"
  >
    <div
      class="design-event position-absolute"
    >
      <span
        aria-label="Modified in this version"
        title="Modified in this version"
      >
        <icon-stub
          cssclasses="text-primary-500"
          name="file-modified-solid"
          size="18"
        />
      </span>
    </div>
     
    <img
      alt="test"
      class="block ml-auto mr-auto mw-100 mh-100 design-img"
      src="http://via.placeholder.com/300"
    />
  </div>
   
  <div
    class="card-footer d-flex w-100"
  >
    <div
      class="d-flex flex-column str-truncated-100"
    >
      <span
        class="bold str-truncated-100"
      >
        test
      </span>
       
      <span
        class="str-truncated-100"
      >
        
        Updated 
        <timeago-stub
          cssclass=""
          time="01-01-2019"
          tooltipplacement="bottom"
        />
      </span>
    </div>
     
    <!---->
  </div>
</router-link-stub>
`;

exports[`Design management list item component renders item with multiple comments 1`] = `
<router-link-stub
  class="card cursor-pointer text-plain js-design-list-item design-list-item"
  to="[object Object]"
>
  <div
    class="card-body p-0 d-flex align-items-center overflow-hidden position-relative"
  >
    <!---->
     
    <img
      alt="test"
      class="block ml-auto mr-auto mw-100 mh-100 design-img"
      src="http://via.placeholder.com/300"
    />
  </div>
   
  <div
    class="card-footer d-flex w-100"
  >
    <div
      class="d-flex flex-column str-truncated-100"
    >
      <span
        class="bold str-truncated-100"
      >
        test
      </span>
       
      <span
        class="str-truncated-100"
      >
        
        Updated 
        <timeago-stub
          cssclass=""
          time="01-01-2019"
          tooltipplacement="bottom"
        />
      </span>
    </div>
     
    <div
      class="ml-auto d-flex align-items-center text-secondary"
    >
      <icon-stub
        class="ml-1"
        cssclasses=""
        name="comments"
        size="16"
      />
       
      <span
        aria-label="2 comments"
        class="ml-1"
      >
        
        2
      
      </span>
    </div>
  </div>
</router-link-stub>
`;

exports[`Design management list item component renders item with no status icon for none event 1`] = `
<router-link-stub
  class="card cursor-pointer text-plain js-design-list-item design-list-item"
  to="[object Object]"
>
  <div
    class="card-body p-0 d-flex align-items-center overflow-hidden position-relative"
  >
    <!---->
     
    <img
      alt="test"
      class="block ml-auto mr-auto mw-100 mh-100 design-img"
      src="http://via.placeholder.com/300"
    />
  </div>
   
  <div
    class="card-footer d-flex w-100"
  >
    <div
      class="d-flex flex-column str-truncated-100"
    >
      <span
        class="bold str-truncated-100"
      >
        test
      </span>
       
      <span
        class="str-truncated-100"
      >
        
        Updated 
        <timeago-stub
          cssclass=""
          time="01-01-2019"
          tooltipplacement="bottom"
        />
      </span>
    </div>
     
    <!---->
  </div>
</router-link-stub>
`;

exports[`Design management list item component renders item with single comment 1`] = `
<router-link-stub
  class="card cursor-pointer text-plain js-design-list-item design-list-item"
  to="[object Object]"
>
  <div
    class="card-body p-0 d-flex align-items-center overflow-hidden position-relative"
  >
    <!---->
     
    <img
      alt="test"
      class="block ml-auto mr-auto mw-100 mh-100 design-img"
      src="http://via.placeholder.com/300"
    />
  </div>
   
  <div
    class="card-footer d-flex w-100"
  >
    <div
      class="d-flex flex-column str-truncated-100"
    >
      <span
        class="bold str-truncated-100"
      >
        test
      </span>
       
      <span
        class="str-truncated-100"
      >
        
        Updated 
        <timeago-stub
          cssclass=""
          time="01-01-2019"
          tooltipplacement="bottom"
        />
      </span>
    </div>
     
    <div
      class="ml-auto d-flex align-items-center text-secondary"
    >
      <icon-stub
        class="ml-1"
        cssclasses=""
        name="comments"
        size="16"
      />
       
      <span
        aria-label="1 comment"
        class="ml-1"
      >
        
        1
      
      </span>
    </div>
  </div>
</router-link-stub>
`;
